tests/datasource/fluent/test_metadatasource.py

```diff
@@ -1,24 +1,106 @@
+from __future__ import annotations
+
 import copy
 import inspect
-from pprint import pformat as pf
-from typing import ClassVar, List, Optional, Type
+from typing import List, Type
 
 import pytest
 
 from great_expectations.execution_engine import ExecutionEngine
-from great_expectations.experimental.context import get_context
-from great_expectations.experimental.datasources.interfaces import (
+from great_expectations.datasource.fluent.interfaces import (
     BatchRequest,
     BatchRequestOptions,
     DataAsset,
     Datasource,
 )
-from great_expectations.experimental.datasources.metadatasource import MetaDatasource
-from great_expectations.experimental.datasources.sources import (
+from great_expectations.datasource.fluent.metadatasource import MetaDatasource
+from great_expectations.datasource.fluent.sources import (
     TypeRegistrationError,
     _SourceFactories,
 )
 
+import logging
+import pathlib
+from pprint import pformat as pf
+from typing import ClassVar, Dict, Optional, Union
+
+from pydantic import DirectoryPath, validate_arguments
+
+from great_expectations.datasource.fluent.config import GxConfig
+
+
+logger = logging.getLogger(__name__)
+
+
+# Fake DataContext for testing
+class DataContext:
```

Comment:
This `DataContext` use to live the `great_expectations/experimental/data_context.py` but it is only used for testing in this file so I moved it here.